staging/src/k8s.io/client-go/rest/transport.go

```diff
@@ -95,5 +95,6 @@ func (c *Config) TransportConfig() (*transport.Config, error) {
 			Groups:   c.Impersonate.Groups,
 			Extra:    c.Impersonate.Extra,
 		},
+		TokenSource: c.BearerTokenSource,
```

Comment:
since auth providers can return a bearer token source, I would have expected this to be initialized from the auth provider above